### PR TITLE
Create LLVM output

### PR DESCRIPTION
**Pull request by Zalinafali**
Create basic LLVM output to file:
- empty AST
- BinOp (+,-,*,/))
- Variable
- Value

### CODE REVIEW

File: nodes.py [hunk 1]
```diff
@@ -217,3 +319,54 @@ def check_semantic_errors(self):
                 semantic_check, _ = node.check_semantics(variables_dict)
                 if semantic_check != 0:
                     return 1
+                
+    def create_llvm_output(self, filename):
+        variables_dict = dict() #TODO check if ProgramMemory works correctly
+        output_lines = []
+        initiation_section = True
```
Comment (j-janiszewski) [inline nodes.py]: This variable is not used at all, why we declare it and modify it's value later?

File: nodes.py [hunk 2]
```diff
@@ -171,6 +270,9 @@ def __str__(self, indent_level=0) -> str:
             indent_level, f"(value: {self.value}, value_type: {self.value_type})"
         )
 
+    def write_code(self, output_lines):
+        return self.value_type, -1, self.value #TODO check
```
Comment (j-janiszewski) [inline nodes.py]: check what?

File: nodes.py [hunk 3]
```diff
@@ -217,3 +319,54 @@ def check_semantic_errors(self):
                 semantic_check, _ = node.check_semantics(variables_dict)
                 if semantic_check != 0:
                     return 1
+                
+    def create_llvm_output(self, filename):
+        variables_dict = dict() #TODO check if ProgramMemory works correctly
```
Comment (j-janiszewski) [inline nodes.py]: What's the relation between this variables_dict and the one from ProgramMemory?

File: nodes.py [hunk 4]
```diff
@@ -154,6 +226,33 @@ def __str__(self, indent_level=0):
...
+        elif self.variable_type is Types.Float:
+            output_lines.append(f"  %{ProgramMemory.mem_counter} = alloca float, align 4")
+            ProgramMemory.mem_counter += 1
+        elif self.variable_type is Types.Bool:
+            output_lines.append(f"  %{ProgramMemory.mem_counter} = alloca i1")
+            ProgramMemory.mem_counter += 1
+        return
+    
+    def write_code(self, output_lines):
+        #TODO dopisać
```
Comment (j-janiszewski) [inline nodes.py]: What exactly  "dopisać" ; )?

File: nodes.py [hunk 5]
```diff
@@ -80,6 +87,71 @@ def __handle_arithmetic_operator(self, operation_name, left_type, right_type):
 
     def __str__(self, indent_level=0):
         return super().__str__(indent_level, f"({self.op})")
+    
+    def write_code(self, output_lines):
+        if self.op in ["+", "-", "*", "/"]:
+            left_type, left_mem_id, left_val = self.left.write_code(output_lines)
+            right_type, right_mem_id, right_val = self.right.write_code(output_lines)
+            if isinstance(self.left, Assign):
+                print(f"BinOp -> Assign TODO")  #TODO Assign
```
Comment (j-janiszewski) [inline nodes.py]: Assign node is never a left node of BinOP node. Check how Assignement is created.

File: nodes.py (hunk 4 above)
Comment (Zalinafali) [reply nodes.py]: Nothing, it's some leftover. Comment to delete.

File: nodes.py (hunk 5 above)
Comment (Zalinafali) [reply nodes.py]: True, fixing with Assign write_code change.

File: nodes.py (hunk 2 above)
Comment (Zalinafali) [reply nodes.py]: Check if returned values are proper for Value node - they are, comment to delete.

File: nodes.py (hunk 3 above)
Comment (Zalinafali) [reply nodes.py]: Dictionary moved to ProgramMemory, this variable is unnecessary.

File: nodes.py (hunk 1 above)
Comment (Zalinafali) [reply nodes.py]: Not needed at the moment, to delete.

**Pull request merged**